useragent/user_agent.go

```diff
@@ -79,19 +86,48 @@ type info struct {
 	Value string
 }
 
+// Sanitize the user agent value. This is useful when the value is not ensured to be
+// to be valid at compile time. Having this sanitization then ensures downstream
+// applications can correctly parse the full user agent header, by making sure
+// characters like '/' and ' ' are not present in the value.
+func Sanitize(s string) string {
+	allowList := func(r rune) rune {
+		if r == '.' || unicode.IsLetter(r) || unicode.IsDigit(r) {
```

Comment:
This includes everything in the "letter" class, which includes modifiers, accents, and more that doesn't match the regexp for alphanumerical values.

To guarantee that sanitize produces a valid value it would need to use the same set of characters as the characters in the alphanumeric regexp and map all non-matching characters to the placeholder.

Reply:
Good catch. Did some research and it seems UTF8 characters are not supported in general in HTTP headers. The recommended approach seems to be to stick to ASCII characters in the HTTP headers. 

I have changed the validation to allow alphanumeric, `-`, `_` AND `.` as valid characters.